starknet-replay/src/runner/replace_ids.rs

```diff
@@ -150,6 +154,55 @@ mod tests {
         fs::read_to_string(sierra_program_json_file)
     }
 
+    // This is because the built-in equality doesn't check for matching `debug_name`
+    // string.
+    fn compare_libfunc_declaration(a: &[LibfuncDeclaration], b: &[LibfuncDeclaration]) -> bool {
+        if a.len() != b.len() {
+            return false;
+        }
+        for (a, b) in a
+            .iter()
+            .sorted_by(|a, b| Ord::cmp(&a.id.id, &b.id.id))
+            .zip(b.iter().sorted_by(|a, b| Ord::cmp(&a.id.id, &b.id.id)))
+        {
+            if a.id.id != b.id.id {
+                println!("Different ids. Expected {:?} | Actual {:?}", a.id, b.id);
+                return false;
+            }
+            if a.id.debug_name != b.id.debug_name {
+                println!(
+                    "Different debug_name. Expected {:?} | actual {:?}",
+                    a.id, b.id
+                );
+                return false;
+            }
+        }
+        true
+    }
+
+    // This is because the built-in equality doesn't check for matching `debug_name`
+    // string.
+    fn compare_type_declaration(a: &[TypeDeclaration], b: &[TypeDeclaration]) -> bool {
```

Comment:
Compare to me implies returning an ordering. I think this is an equality check. I suggest renaming to `type_declaration_eq`.